terraform-provider-constellation/internal/provider/attestation_data_source_test.go

```diff
@@ -16,6 +17,21 @@ func TestAccAttestationSource(t *testing.T) {
 	// Set the path to the Terraform binary for acceptance testing when running under Bazel.
 	bazelPreCheck := func() { bazelSetTerraformBinaryPath(t) }
 
+	assertNonZeroValue := func(attr string) resource.TestCheckFunc {
+		return resource.TestMatchResourceAttr(
+			"data.constellation_attestation.test",
+			attr,
+			regexp.MustCompile(`^[1-9]\d*$`),
```

Comment:
Could we rather use https://pkg.go.dev/github.com/hashicorp/terraform-plugin-testing/helper/resource#TestCheckResourceAttrWith? I fear that regexp matching of numbers is somewhat error prone.

Reply:
I guess everyone hates regex :D it matches only digits. How is this error prone?
But I can also rewrite the logic in Go.

Reply:
This one does not match `012`, which may or may not be a valid uint8, depending on who you ask. The other one matches `foo1bar`, which is too permissive I believe.

I'm not against regexps per se, but https://regex.info/blog/2006-09-15/247.